Assignment 3

TilegenGena opened the pull request.

TilegenGena committed:
Install Angular and create new Angular app

Summary:
1. Install angular project and named 'client'
	* disabled autocomplection
	* shared usage data
	* enabled angular routing
	* choose SCSS format styles
2. Copied files to `.dockerignore` from `client/.gitignore`

Test:
Angular installed successfulyy in client dir

networkflow reviewed (changes requested):
* Could you change the "base" branch for this PR to `new-branch` (the one you're using for assignment 2) instead of `main`? (The "Opening a pull request" section in https://quip.com/Qp1LAMQR2g2S/Instructions-for-Submitting-Tutorial-App-Pull-Requests has more info with a screenshot of how to do this.)
* Could you add a test plan to the commit message?
  * Also, your commit message should start with a title on the first line, and then you can write the summary. For example:
```
My Title

Summary:
* Summary point 1

Test plan:
* Test plan point 1
```
* I think there's one more commit for assignment 3 (serving Angular from NestJS). I'll wait to fully review this until that commit's ready, but this looks good so far!

TilegenGena committed:
Serving Angular client app from NestJS

Summary:
	* Install serve static module used npm command  `npm install --save @nestjs/serve-static`
	* Delete `getHello`
	* Connected my angular with server. Set up rootPath to client file at the `app.module.ts`
	* Update docker file add commands:
		WORKDIR /app/client
		RUN npm ci
		RUN npm install -g @angular/cli
		RUN ng build

		WORKDIR /app/server
		RUN npm ci
		RUN npm run build

Test:
client dir run:
	 `ng build --watch`
server dir run:
	`npm run start:prod`

	Expected:
	Local port showin Angular page

networkflow reviewed (changes requested):
This PR includes the commit from assignment 2: requesting changes for you to rebase this branch on `main` so it won't include that commit, but once you do I can review it 🙂 Let me know if you have any questions about fixing the Git issue!

networkflow commented on .dockerignore:
```diff
@@ -1,3 +1,47 @@
+# Compiled output
+client/dist
+client/tmp
+client/out-tsc
+client/bazel-out
+
+# Node
+client/node_modules
+client/npm-debug.log
+client/yarn-error.log
+
+# IDEs and editors
+client/.idea/
+client/.project
+client/.classpath
+client/.c9/
+client/*.launch
+client/.settings/
+client/*.sublime-workspace
+
+# Visual Studio Code
+client/.vscode/*
+!client/.vscode/settings.json
+!client/.vscode/tasks.json
+!client/.vscode/launch.json
+!client/.vscode/extensions.json
+client/.history/*
+
+# Miscellaneous
+client/.angular/cache
+client/.sass-cache/
+client/connect.lock
+client/coverage
+client/libpeerconnection.log
+client/testem.log
+client/typings
+
+# System files
+client/.DS_Store
+client/Thumbs.db
```
Nice, looks like you did a good job copying all these files over!

networkflow commented on dockerfile:
```diff
@@ -1,11 +1,13 @@
+WORKDIR /app/server
+RUN npm install
+RUN npm build
 
```
It's ok for now, but in the real world you definitely don't want to change important code like the `Dockerfile` without mentioning it in the `Summary` and also explaining how you tested it in the `Test Plan`. I've seen things break at Facebook because someone did something they thought was a simple, safe change but they didn't test it enough and it caused big problems 🙂

networkflow commented on client/src/app/app.component.html:
```diff
@@ -1,4 +1,4 @@
-<!-- * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * -->
+* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
```
I think this change wasn't intended?

networkflow commented on server/src/app.module.ts:
```diff
@@ -1,9 +1,19 @@
+import { join } from 'path';
 
+// dirname = /server/dist 
+console.log(__dirname);
+
+//  goal = /client/dist/client
```
This is helpful for debugging, but could you remove it now? (You could leave the `__dirname = /server/dist` comment if you want, it may be helpful for other people reading the code to understand what `__dirname` is expected to be.)

networkflow reviewed (approved):
Overall looks 💯, nice work!

First commit:
* Nice summary!
* For the test plan, saying Angular was installed successfully is a little vague (how did you know it was installed successfully?). I would suggest being more specific, in this case maybe saying something like, "It doesn't affect the app for now, but I tested Angular in the next commit and the installation worked."

Second commit:
* For the summary, you don't need to copy-paste in the Docker commands 🙂 You can just briefly describe what you changed ("Added Docker instructions to build the the client code") and reviewers can see the exact commands in the code below
* Thanks for including specific commands in the test plan!
